### PR TITLE
WIP documenting activity endpoint

### PR DESCRIPTION
**Pull request by geoffwhittington**

### CODE REVIEW

File: source/includes/activity.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Content-Type: application/json
+
+{
+    "results": [{
+        "id": "3",
+        "date": "2016-03-30T17:10:20.532634+00:00",
+        "user": {"id": "7", "email": "cindy@example.com"},
+        "type": "ModifiedBusinessUnitActivity",
+        "text": "<i>Admin Testerton</i> modified business unit \n    <a href="/bunits/hotline-bling-500/apps/" title="Hotline Bling 500">Hotline Bling 500</a>\n\n",
+        "data": {"business_unit": "5"}
```
Comment (geoffwhittington) [inline source/includes/activity.md]: We need to fix SDE-1680 to actually use business_unit.

Comment (geoffwhittington) [inline source/includes/activity.md]: What happens when the business unit is deleted - what do we get for the value?

File: source/includes/activity.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "results": [{
+        "id": "3",
+        "date": "2016-03-30T17:10:20.532634+00:00",
+        "user": {"id": "7", "email": "cindy@example.com"},
+        "type": "ModifiedBusinessUnitActivity",
+        "text": "<i>Admin Testerton</i> modified business unit \n    <a href="/bunits/hotline-bling-500/apps/" title="Hotline Bling 500">Hotline Bling 500</a>\n\n",
```
Comment (geoffwhittington) [inline source/includes/activity.md]: Is the formatting conditional on the user sending "to_html=True" in the parameters like what happens with the `/tasks` endpoint?

File: source/includes/activity.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Activity
+
+## Get all activity resources
+
+```http
+GET /api/v2/activity/ HTTP/1.1
```
Comment (geoffwhittington) [inline source/includes/activity.md]: `GET /api/v2/activity/?to_html=True` for the HTML formatting?

File: source/includes/activity.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "results": [{
+        "id": "3",
+        "date": "2016-03-30T17:10:20.532634+00:00",
+        "user": {"id": "7", "email": "cindy@example.com"},
```
Comment (geoffwhittington) [inline source/includes/activity.md]: Should user just be an *expand* attribute and we only show `"user": 7` here. If the consumer sends `/api/v2/activity-log/?expand=user` we'd get all of the user attributes.

**Comment by geoffwhittington** [comment]
Should the endpoint allow consumers to request a specific activity i.e. `/api/v2/activity-log/1/` ? If so we should pluralize the name the endpoint. Is it "restful" to only provide a list?

File: source/includes/activity.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        "text": "<i>Admin Testerton</i> modified business unit \n    <a href="/bunits/hotline-bling-500/apps/" title="Hotline Bling 500">Hotline Bling 500</a>\n\n",
+        "data": {"business_unit": "5"}
+    },
+    {
+        "id": "2",
+        "date": "2016-03-30T15:16:05.659610+00:00",
+        "user": {"id": "1", "email": "admin@example.com"},
+        "type": "UserCreatedActivity",
+        "text": "<i>Admin Testerton</i> created user <i>Cindy Lu</i>",
+        "data": {"created_user": "7"}
```
Comment (geoffwhittington) [inline source/includes/activity.md]: Should be `"created_user": 7`

File: source/includes/activity.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        "date": "2016-03-30T17:10:20.532634+00:00",
+        "user": {"id": "7", "email": "cindy@example.com"},
+        "type": "ModifiedBusinessUnitActivity",
+        "text": "<i>Admin Testerton</i> modified business unit \n    <a href="/bunits/hotline-bling-500/apps/" title="Hotline Bling 500">Hotline Bling 500</a>\n\n",
+        "data": {"business_unit": "5"}
+    },
+    {
+        "id": "2",
+        "date": "2016-03-30T15:16:05.659610+00:00",
+        "user": {"id": "1", "email": "admin@example.com"},
```
Comment (geoffwhittington) [inline source/includes/activity.md]: `"id": 1`

**Comment by geoffwhittington** [comment]
@lmontopo I left some comments. Looking good!

File: source/includes/activity.md (hunk 2 above)
Comment (lmontopo) [inline source/includes/activity.md]: My understanding from our discussions on the ticket was that people wanted the 'text' field to actually BE the html.  Personally, I like your suggestion better. @funkaoshi, thoughts?

File: source/includes/activity.md (hunk 4 above)
Comment (lmontopo) [inline source/includes/activity.md]: I'm not sure.  We aren't really consistent in our api in this respect.  For instance, the tasks endpoint has an 'assigned_to' field, and each user is displayed with both the id and the email.  However, the projects endpoint returns a 'creator' field which is populated with just the user_id, by default.

File: source/includes/activity.md (hunk 5 above)
Comment (lmontopo) [inline source/includes/activity.md]: Noted, thank you.

File: source/includes/activity.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+{
+    "results": [{
+        "id": 3,
+        "date": "2016-03-30T17:10:20.532634+00:00",
+        "actor": 7,
+        "name": "ModifiedBusinessUnitActivity",
+        "text": "Admin Testerton modified business unit Example BU 500",
+        "data": {
+            "business_unit": 5,
+            "business_unit_desc"="Example BU 500"
```
Comment (RouganStriker) [inline source/includes/activity.md]: Should be a colon

File: source/includes/activity.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        "date": "2016-03-30T15:14:46.634929+00:00",
+        "actor": 1,
+        "name": "LoggedInActivity",
+        "text": "Admin Testerton logged in",
+        "data": {}
+    }]
+}
+```
+
+This endpoint returns a list of Activities.
```
Comment (funkaoshi) [inline source/includes/activity.md]: "... events that have occurred in the application, when they occurred, and additional data about the event."

I think one thing we should in our docs is explain in more detail what all the resources are about. We often assume people will know what a Business Unit is or a Application.

File: source/includes/activity.md (hunk 7 above)
Comment (lmontopo) [inline source/includes/activity.md]: ou! thanks!

File: source/includes/activity.md (hunk 8 above)
Comment (lmontopo) [inline source/includes/activity.md]: good idea.  I'll make that change.

File: source/includes/activity.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+### Query Parameters
+
+The following parameters may be used to filter the activity resources in the response.
+
+Parameter        | Description
+-----------------|-------------
+actor            | Returns a list of activities performed by the user with the id specified.
+application      | Returns a list of activities that reference the application with the specified id.
+application_desc | Returns a list of activities that reference the application with the specified applicaiton_desc.
```
Comment (sibartlett) [inline source/includes/activity.md]: Mispelt `applicaiton_desc`.

Comment (lmontopo) [inline source/includes/activity.md]: fixed

**Pull request merged**